tools/generator/templates/player/java/interface.cc.jinja2

```diff
@@ -0,0 +1,338 @@
+        jrt.env->SetObjectArrayElement(out, (jsize)i, cxx_to_java<CxxType, jobject>(in[i]));
+    return out;
+}
+
+template <typename Lang, typename CxxType>
+std::vector<CxxType> java_to_cxx_array(jobject in)
+{
+    jobjectArray array = (jobjectArray)in;
+    size_t size = (size_t)jrt.env->GetArrayLength(array);
+    std::vector<CxxType> out;
+    for (size_t i = 0; i < size; i++)
+        out.push_back(java_to_cxx<jobject, CxxType>(jrt.env->GetObjectArrayElement(array, (jsize)i)));
+    return out;
+}
+{% for enum in game.enum %}
+
+template<>
+{{ enum.enum_name }} java_to_cxx<jobject, {{ enum.enum_name }}>(jobject in)
+{
+    jmethodID ordinal = jrt.env->GetMethodID({{ enum.enum_name|camel_case }}::Class(), "ordinal", "()I");
+    return {{ enum.enum_name }}(java_to_cxx<jint, int>(jrt.env->CallIntMethod(in, ordinal)));
+}
+
+template<>
+jobject cxx_to_java<{{ enum.enum_name }}, jobject>({{ enum.enum_name }} in)
+{
+    jmethodID method = jrt.env->GetStaticMethodID({{ enum.enum_name|camel_case }}::Class(), "values", "()[L{{ enum.enum_name|camel_case }};");
+    jobjectArray values = (jobjectArray)jrt.env->CallStaticObjectMethod({{ enum.enum_name|camel_case }}::Class(), method);
+    return jrt.env->GetObjectArrayElement(values, (jsize)in);
+}
+{% endfor %}
+{% for struct in game.struct %}
+template <>
+{{ struct.str_name }} java_to_cxx<jobject, {{ struct.str_name }}>(jobject in)
+{
+    {{ struct.str_name }} out;
+    {% for field_name, field_type, _ in struct.str_field %}
+    out.{{ field_name }} = {{ field_type|java_to_cxx }}(jrt.env->Get{{ field_type|java_type_method_call }}Field(in, jrt.env->GetFieldID({{ struct.str_name|camel_case }}::Class(), "{{ field_name }}", "{{ field_type|jni_type_signature }}")));
+    {% endfor %}
+    return out;
+}
+
+template <>
+jobject cxx_to_java<{{ struct.str_name }}, jobject>({{ struct.str_name }} in)
+{
+    jobject out = jrt.env->NewObject({{ struct.str_name|camel_case }}::Class(), jrt.env->GetMethodID({{ struct.str_name|camel_case }}::Class(), "<init>", "()V"));
+    {% for field_name, field_type, _ in struct.str_field %}
+    jrt.env->Set{{ field_type|java_type_method_call }}Field(out, jrt.env->GetFieldID({{ struct.str_name|camel_case }}::Class(), "{{ field_name }}", "{{ field_type|jni_type_signature }}"), {{ field_type|cxx_to_java }}(in.{{field_name }}));
+    {% endfor %}
+    return out;
+}
+{% endfor %}
+{% for func in game.function %}
+
+{{ func.fct_ret_type|jni_type }} {{ func.fct_name }}(JNIEnv* _env, jobject _obj
+{%- for arg_name, arg_type, _ in func.fct_arg -%}
+, {{ arg_type|jni_type }} {{ arg_name }}
+{%- endfor %})
+{
+    {%+ if func is returning %}return {{ func.fct_ret_type|cxx_to_java }}{% endif -%}
+    (api_{{ func.fct_name }}(
+    {%- set comma = joiner(', ') -%}
+    {%- for arg_name, arg_type, _ in func.fct_arg -%}
+    {{ comma() }}{{ arg_type|java_to_cxx }}({{ arg_name }})
+    {%- endfor %}));
+}
+{% endfor %}
+{% for enum in game.enum %}
```

Comment:
I think we said we were going to generate the afficher_* functions directly in game.py

Reply:
I agree, but while it is not done at least the generator is complete.